GVFS/GVFS.Common/Tracing/JsonTracer.cs

```diff
@@ -83,13 +90,22 @@ public bool HasLogFileEventListener
 
         public void AddEventListener(EventListener listener)
         {
+            if (Volatile.Read(ref this.isDisposed) == DISPOSED_TRUE)

```

Comment:
Another question - we have the `Volatile` / `Interlocked` actions on this variable, but I think we can still have race conditions. What if we:

1) Enter `AddEventListener(...)`
2)    Pass the check for `isDisposed` (and we are not disposed), but immediately after is interrupted 
3) Another method disposes the Tracer, and completes the dispose method
4) The rest this method executes - but the object is disposed.